### PR TITLE
mgr/nfs: more tests; add --port to 'nfs cluster create'

### PR DESCRIPTION
**Pull request by liewegas**
- add --port to 'nfs cluster create' command and 'nfs cluster info'
- separate test_nfs out from test_orch_cli
- add test that takes backend ganeshas offline and waits for cephadm to redeploy+reconfigure

Note that the ganesha test does a 'ceph orch ps --refresh' command because the cephadm reconcile loop is pretty slow. Someday soon this will be fixed.

## Checklist
- [ ] References tracker ticket
- [ ] Updates documentation if necessary
- [ ] Includes tests for new functionality or reproducer for bug

---

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`

</details>

### CODE REVIEW

**Review by batrick — COMMENTED** [review]
`add --port to 'nfs cluster create' command and 'nfs cluster info'`

Is it necessary that all NFS gateways in the same cluster listen on the same port? I wonder if what we really want here is `--random-port`?

**Comment by liewegas** [comment]
> `add --port to 'nfs cluster create' command and 'nfs cluster info'`
> 
> Is it necessary that all NFS gateways in the same cluster listen on the same port? I wonder if what we really want here is `--random-port`?

In the case where there is a separate ingress service, the backend ganeshas should have dynamically assigned and/or random ports, but cephadm doesn't do that yet.  The idea is to get there eventually.  The --port specified here, though, would be the front-end user-visible port, which I think should remain well-defined.  There may be a use-case for a random user-visible port, but it hasn't come up yet...

File: src/pybind/mgr/nfs/cluster.py [hunk 1]
```diff
@@ -162,11 +175,13 @@ def _show_nfs_cluster_info(self, cluster_id: str) -> Dict[str, Any]:
                         "ip": ip,
                         "port": cluster.ports[0] if cluster.ports else None
```
Comment (varshar16) [inline src/pybind/mgr/nfs/cluster.py:176]: We already have port here.

File: qa/tasks/cephfs/test_nfs.py [hunk 2]
```diff
@@ -465,6 +465,7 @@ def test_cluster_info(self):
                         "port": 2049
```
Comment (varshar16) [inline qa/tasks/cephfs/test_nfs.py:465]: We already have port here.

**Comment by varshar16** [comment]
Please add this tracker ticket: https://tracker.ceph.com/issues/51787 and update the doc.

**Comment by varshar16** [comment]
jenkins test make check

**Review by varshar16 — APPROVED** [review]
Needs qa test for port argument. Otherwise looks good.

**Pull request merged**